### PR TITLE
Corrected incorrect markdown handling added syntax highlighting

### PR DESCRIPTION
**Pull request by eeyun**
This PR changes the markdown handling engine to redcarpet which has a more common syntax handling than kramdown.

Because of this, some of the new tutorial content had to be brought in line with redcarpet's syntax preferences.

It also adds in highlight.js for syntax highlighting over all of our codeblocks and no longer relies on middleman's syntax highlighter.

Signed-off-by: Ian Henry <ihenry@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by eeyun** [comment]
fixes https://github.com/habitat-sh/habitat/issues/2862

**Comment by eeyun** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/262837301).

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/262837301).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/262837301).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

File: www/source/stylesheets/_global.scss [hunk 1]
```diff
@@ -171,7 +171,7 @@ body {
 // 2. Code Overrides
 // ---------
 pre {
-  background-color: $hab-navy;
+  background-color: #2B303B;
```
Comment (mgamini) [inline www/source/stylesheets/_global.scss]: This is pretty close to `$hab-navy-2` - does that not look ok?

**Comment by eeyun** [comment]
It's close. The current setting is a little darker. If we want to set this
back to the bluer color we can modify the ocean css to match the background
color here!

**Comment by apriofrost** [comment]
Thanks for the awesome work, Ian! I have a few feedback on the colors:
1. One thing we might need to pay attention to is the accessibility of the colors, especially for the single line command under the Interactive Demo section (http://www.acceptance.habitat.sh/tutorials/get-started/demo/). Because currently, a user has to read and type in the command manually (we hasn't implemented the copy pasting feature yet). It has to be very easy to read. The color contrast shown in the screenshot below doesn't satisfy that (I haven't tried with a color contrast checker, but it's unlikely to pass). I would suggest that we just use the current color on the production site.
image
2. The background color of the simulated terminal doesn't match with the new background color of the command line. It's better to keep them the same color when they appear close together.  I would prefer to keep it as the current dark blue color (#283c4c) since it matches better with the current visual style across the site.
image

**Comment by eeyun** [comment]
Ah yep this is due to incorrect syntax on the command. If we update the
content to start with a $ then it should autocorrect to appropriate
highlighting

**Comment by apriofrost** [comment]
Cool! I chatted with @davidwrede about this and he currently doesn't have the bandwidth to update the content to include the $ sign. @eeyun Do you think it will be a good idea to suspend on the syntax highlighting for now until we have a better understanding of what color palette works best? This is a great thing to do to improve the readability for our users, just that accessibility and branding need also be taken into consideration.

**Comment by eeyun** [comment]
I can get these pages updated as well. It should be trivial to roll that into this PR. I also don't think we should hold off on syntax highlighting. The code is in and functional and doesn't break anything. I don't mind tweaking the initial colors to get it close to begin with. Afterwards we can come back and iterate on it if we want some really granular and specific palette for the highlights themselves, but I don't think there's a really strong reason to not get the update in and worry about that later. With the exception of the background the highlight colors are very minimal so it shouldn't be an issue.

**Comment by apriofrost** [comment]
Cool! As long as we can replace the gray color appeared in the '--peer' in the screenshot with a color that contrasts well with the background, I'm cool with it. 
If you can make the adjustment to the initial colors, here are some to consider (it's the habitat colors that contrast relatively well with the dark blue background):
#FF9012
#EB6852
#87B09A
#C3C6C8

Thanks so much Ian!

**Comment by eeyun** [comment]
Ok. I've corrected all of the syntax highlighting colors to match the habitat brand colors from brand.chef.io which are the same colors you've posted above @apriofrost . I've also corrected an issue with the new markdown engine that wasn't rendering tables. And I've added correct syntax to the demo pages so the syntax highlighting should be correct there. I'm going to go ahead and push this out to acceptance @apriofrost @cnunciato @mgamini feel free to check it out www.acceptance.habitat.sh would like to get this merged today!

**Comment by eeyun** [comment]
One other note: For the background color of the codeblocks I've switched it to match the deep greenish blue of the demo terminal.

**Comment by eeyun** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/263106349).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/263106349).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by eeyun** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/263176934).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/263176934).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**